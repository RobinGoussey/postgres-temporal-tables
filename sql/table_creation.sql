CREATE EXTENSION IF NOT EXISTS psql_temporal;

CREATE TABLE firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('firemen_temporal');

-- check that history table is created exist
SELECT EXISTS (
   SELECT FROM information_schema.tables 
   WHERE  table_schema = 'temporal_firemen_temporal_history'
   AND    table_name   = 'table_name'
   );

-- make sure the previous history is transferred.
select count(id) from temporal_firemen_temporal_history;