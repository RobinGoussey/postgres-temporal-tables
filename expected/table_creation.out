CREATE EXTENSION IF NOT EXISTS psql_temporal;
CREATE TABLE firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('firemen_temporal');
NOTICE:  trigger "firemen_temporal_temporal_trigger" for relation "public.firemen_temporal" does not exist, skipping
 convert_to_temporal 
---------------------
 t
(1 row)

-- check that history table is created exist
SELECT EXISTS (
   SELECT FROM information_schema.tables 
   WHERE  table_schema = 'temporal_firemen_temporal_history'
   AND    table_name   = 'table_name'
   );
 exists 
--------
 f
(1 row)

-- make sure the previous history is transferred.
select count(id) from temporal_firemen_temporal_history;
 count 
-------
     1
(1 row)

